src/test/java/net/snowflake/ingest/FipsCompliantIngestIT.java

```diff
@@ -0,0 +1,328 @@
+package net.snowflake.ingest;
+
+import net.snowflake.ingest.connection.HistoryRangeResponse;
+import net.snowflake.ingest.connection.HistoryResponse;
+import net.snowflake.ingest.connection.IngestResponse;
+import net.snowflake.ingest.connection.IngestResponseException;
+import net.snowflake.ingest.connection.IngestStatus;
+import net.snowflake.ingest.example.IngestExampleHelper;
+import net.snowflake.ingest.fips.BouncyCastleFIPSInit;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.spec.RSAKeyGenParameterSpec;
+import java.sql.Connection;
+import java.time.Instant;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * This test is specifically testing REST apis with FIPS compliance mode
+ * enabled.
+ * <p>
+ * <b>Prerequisites:</b>
+ * You must have a valid set of Snowflake credentials
+ * We use encrypted key pair so follow the instructions written on the
+ * documentation and also in readme of this repo.
+ */
+public class FipsCompliantIngestIT
+{
+  // Details required to connect to Snowflake over jdbc and establish
+  // pre-requesites
+  private static final String ACCOUNT = "s3testaccount";
```

Comment:
Will this test now run as part of the Travis tests?

Reply:
Apparently yes and is is failing. I am investigating that. Also, I don't think this test should be part of Travis. (If you think otherwise), I will figure out how to exclude this.

Reply:
Agree - it is a very specific test that we dont need to run frequently.